golem-ts-agentic/packages/golem-ts-sdk/src/index.ts

```diff
@@ -100,16 +101,107 @@ async function invoke(
 }
 
 async function discoverAgentTypes(): Promise<bindings.guest.AgentType[]> {
-  return AgentTypeRegistry.getRegisteredAgents();
+  try {
+    return AgentTypeRegistry.getRegisteredAgents();
+  } catch (e) {
+    // Have to throw AgentError, as the discover-agent-types WIT function returns result<list<agent-type>, AgentError>
+    if (isAgentError(e)) {
+      throw e;
+    } else {
+      throw createCustomError(String(e));
```

Comment:
<img width="153" height="45" alt="Image" src="https://github.com/user-attachments/assets/1904e68e-af65-476a-8343-a659e4eb1c0f" />

i think this should check for:
- undefined
- strings, return them as they are
- try to stringify objects
- then fallback to this

Reply:
but probably currently the most common one is string, that should work even with this